### PR TITLE
permissions docs

### PR DESCRIPTION
**Pull request by Trivo25**
closes [#15](https://github.com/o1-labs/docs2/issues/15#issue-1086221977)

### CODE REVIEW

File: docs/zkapps/advanced-snarkyjs/permissions.mdx [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+Currently, we have 5 types of authorizations: `none`, `impossible`, `signature`, `proof` and `proofOrSignature`.
+
+We already know that a transaction consists of multiple account updates (sort of like instructions to the network) - and each account update has to be authorized in one way or another.
+When you inspect an account update (either directly in SnarkyJS or using an explorer), you will notice a field `authorization` on the account update.
+If this field has a proof attached, it means it's authorized by a proof, which will be checked against the verification key of the account. If it's a signature, that means the account update was authorized by a signature.
+
+Here is an example of how this may look like: 
+
+```json
+{
```
Comment (Trivo25) [inline docs/zkapps/advanced-snarkyjs/permissions.mdx:74]: I think its important to briefly mention the structure (where does the authorization (sig, proof) come from? how is this enforced?)

File: docs/zkapps/advanced-snarkyjs/permissions.mdx [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+`impossible`: If a fields permission is set to `impossible`, nothing can ever change this field!  
+
+`signature`: Fields which have their permission set to `signature` can only be manipulated by account updates that are accompanied and authorized by a valid signature.
+
+`proof`: Fields which have their permission set to `proof` can only be manipulated by account updates that are accompanied and authorized by a valid proof. Proofs are generated by proving the execution of a smart contract method.
+
+`proofOrSignature`: As the name might suggest, permissions with authorization set to `proofOrSignature` accept both a valid signature or a valid proof. 
+
+## The Default
```
Comment (Trivo25) [inline docs/zkapps/advanced-snarkyjs/permissions.mdx:117]: imo very important to mention that smart contracts have a default

File: docs/zkapps/advanced-snarkyjs/permissions.mdx [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+## Example
+
+To better understand how we can leverage permissions to make our smart contract more secure, let's look at a few examples.
+
+### Sending MINA from smart contracts
+
+Some smart contracts will not only manage state, but also token (such as the native MINA token). In order to prevent malicious actors from withdrawing all funds, we can use permissions to secure them.
+
+Consider the following smart contract `UnsecureContract`, which is available [here](https://github.com/o1-labs/snarkyjs/blob/main/src/examples/zkapps/simple_zkapp_payment.ts) as well
```
Comment (Trivo25) [inline docs/zkapps/advanced-snarkyjs/permissions.mdx:150]: we have lengthy examples that utilize permissions. I wanted to show some code to guide the reader on how to use and define permissions, but i didnt want to write a whole new examples that we would have to maintain, including all boilerplate code etc. thoughts?

**Review by mitschabaude — COMMENTED** [review]
The two new permissions that were added with the latest release are missing - `access` and `setTiming`

File: docs/zkapps/advanced-snarkyjs/permissions.mdx [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+
+Now that we have provided a valid signature and the smart contract requires a valid signatures for sending funds from the smart contract, our transaction will succeed!
+
+
+### Upgradeability of smart contracts
+
+Another important part of smart contract development is upgradeability.
+By using permissions, we can either make our smart contract not upgradeable at all or upgradeable! 
+On Mina, when we deploy a smart contract we generate a verification key from our contract source code, which will then be stored on-chain and used to verify proofs that belong to that smart contract.
+We remember that one permission was called `setVerificationKey` - by modifying the authorization kind for this permission, we can make our smart contract upgradeable or not.
```
Comment (mitschabaude) [inline docs/zkapps/advanced-snarkyjs/permissions.mdx:207]: It's great that you added the upgradability section! I think, though, that the example should also set `setPermissions` to impossible. Only that will make the contract truly frozen

File: docs/zkapps/how-to-write-a-zkapp.mdx [hunk 5]
```diff
@@ -916,7 +916,9 @@ To summarize, there are three types of authorization that account updates can ha
 
 The list above just reflects common defaults. The full source of truth is given by the _account permissions_, see [Permissions](snarkyjs-reference/Interfaces/Permissions). Using permissions, account owners can decide on a fine-grained level which type of authorization is required on which kinds of updates.
 
-<!-- TODO: link to advanced permissions section instead of just API reference; this section should list all permissions, explain how permissions are changed through zkApp deployment, explain how accounts can be made non-upgradeable, and explain the `access` permission for the case of token contracts -->
+If you are looking for an advanced introduction to Permissions, take a look at [this](advanced-snarkyjs/permissions) section!
```
Comment (mitschabaude) [inline docs/zkapps/how-to-write-a-zkapp.mdx:919]: This feels a bit redundant, I would merge this sentence into the paragraph above. For example

```diff
- The full source of truth is given by the _account permissions_, see [Permissions](snarkyjs-reference/Interfaces/Permissions).
+ The full source of truth is given by the _account permissions_, see [our in-depth section about Permissions](advanced-snarkyjs/permissions).
Using permissions, account owners can decide on a fine-grained level which type of authorization is required on which kinds of updates.

- If you are looking for an advanced introduction to Permissions, take a look at [this](advanced-snarkyjs/permissions) section!
```

Also IMO marking links with a generic small word like [this](https://github.com/o1-labs/docs2/pull/277/files) should be avoided

**Review by mitschabaude — COMMENTED** [review]
small comment on the added link:

**Review by mitschabaude — APPROVED** [review]
approving in advance with the two comments above!

File: docs/zkapps/advanced-snarkyjs/permissions.mdx [hunk 6]
```diff
@@ -246,6 +246,9 @@ Using the `LocalBlockchain`, we will get the following (expected) error:
 
 `Error: Transaction verification failed: Cannot update field 'verificationKey' because permission for this field is 'Impossible'`
 
+For the sake of security, it is important to mention that if we want to make our smart contract truly impossible to upgrade, we should also set the field `setPermissions` to `impossible`,
+in order to prevent a malicious actor from changing the permission `setVerificationKey` to, for example, `none` - which would allow an actor to manipulate the verification key again.
```
[the remark this answers is not in the document]
Comment (mitschabaude) [reply docs/zkapps/advanced-snarkyjs/permissions.mdx:253]: or maybe "malicious zkApp developer".. but they don't really have to be malicious for that to be an issue

**Comment by jasongitmail** [comment]
As discussed at Product Eng meeting, skipping readability review and creating a new issue for that to be done async in ~1 week by new tech writer, but we want to get this live asap.

**Pull request merged**